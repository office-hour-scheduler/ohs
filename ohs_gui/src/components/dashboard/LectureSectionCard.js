import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

class LectureSectionCard extends PureComponent {
  render() {
    const { section } = this.props;
    return (
      <Link
        to={{
          pathname: '/section',
          search: `?course=${section.course.courseCode}&year=${
            section.year
          }&semester=${section.semester}&sectionCode=${section.sectionCode}`
        }}
      >
        <div className="card-element lecture-section-verbose">
          <div>
            name: {section.course.courseCode}
            code: {section.sectionCode}
            year: {section.year}
            semester: {section.semester}
            students: {section.numStudents}
          </div>
        </div>
      </Link>
    );
  }
}

LectureSectionCard.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  section: PropTypes.object.isRequired
};

export default LectureSectionCard;
