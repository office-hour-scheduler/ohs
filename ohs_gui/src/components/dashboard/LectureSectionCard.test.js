import React from 'react';
import { shallow } from 'enzyme';

import LectureSectionCard from './LectureSectionCard';

const SECTION = {
  course: {
    courseCode: 'CSC302'
  },
  year: 2019,
  semester: 'FALL',
  sectionCode: 'LEC0101',
  numStudent: 25
};

it('should render correctly with no props', () => {
  const t = () => {
    shallow(<LectureSectionCard />);
  };
  expect(t).toThrow(Error);
});

it('test if it renders with params', () => {
  const component = shallow(<LectureSectionCard section={SECTION} />);
  expect(component).toMatchSnapshot();
});
