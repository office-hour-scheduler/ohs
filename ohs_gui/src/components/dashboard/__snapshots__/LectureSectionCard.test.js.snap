// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if it renders with params 1`] = `
<Link
  replace={false}
  to={
    Object {
      "pathname": "/section",
      "search": "?course=CSC302&year=2019&semester=FALL&sectionCode=LEC0101",
    }
  }
>
  <div
    className="card-element lecture-section"
  >
    <div>
      CSC302H1F - LEC0101 - 2019
    </div>
  </div>
</Link>
`;
